pkg/ipcache/ipcache.go

```diff
@@ -242,24 +263,48 @@ func ipIdentityWatcher(owner IPIdentityMappingOwner) {
 					envoy.NetworkPolicyHostsCache.Upsert(envoy.NetworkPolicyHostsTypeURL, ipIDPair.ID.StringID(), &envoyAPI.NetworkPolicyHosts{Policy: uint64(ipIDPair.ID), HostAddresses: ipStrings}, false)
 				}
 			case kvstore.EventTypeDelete:
+				// Synchronize local caching of endpoint IP to ipIDPair mapping with
+				// operation key-value store has informed us about.
+
+				convertedKey, err := keyToIP(event.Key)
+
+				// Value is not present in deletion event; need to convert key
+				// to IP.
+				if err != nil {
+					scopedLog.Error("error parsing IP from key: %s", err)
+					continue
+				}
+
+				ipIDPair.IP = convertedKey
+				ipStr := convertedKey.String()
+
+				cachedIdentity, exists := IPIdentityCache.LookupByIP(ipStr)
+
 				if exists {
+
+					// Perform lookup first to get list of identities.
+					identityToDelete, _ := IPIdentityCache.LookupByIP(ipStr)
```

Comment:
Is this just `cachedIdentity`? I imagine that's why we don't need to check the exists which I asked about earlier.